user_profile/views.py

```diff
@@ -1,49 +1,30 @@
-from allauth.socialaccount.models import SocialAccount
-from rest_framework import viewsets, permissions
-from rest_framework.generics import get_object_or_404
+from rest_framework import generics, permissions, status
+from rest_framework.request import Request
 from rest_framework.response import Response
 
-from user_profile.models import UserProfile
-from user_profile.serializers import UserProfileSerializer
+from custom_user.models import CustomUser
+from custom_user.serializers import CustomUserSerializer
 
 
-class UserProfileViewSet(viewsets.ModelViewSet):
-    queryset = UserProfile.objects.all()
-    serializer_class = UserProfileSerializer
-    http_method_names = ['get']
+class UserProfileView(generics.RetrieveAPIView,
```

Comment:
Why did you migrated to generics, from viewset?
for readablity?

Reply:
Or get more modifablity ?

Reply:
Viewset is really magic for implement api but for what I want to do, I didn't use it.

We can use generics for detailed implementation that viewsets don't

Reply:
Yep I think viewsets are so abstractic too